### PR TITLE
Add eqBigInt and ordBigInt

### LINKED ISSUE #1371: Add ordBigInt
## 🚀 Feature request

Is there currently an ordering for the `bigint` type? I might be missing something

### Current Behavior

There is no provided ordering for `bigint` 

### Desired Behavior

``` typescript
import { clamp, ordBigInt } from 'fp-ts/Ord'
import { pipe } from 'fp-ts/function'

const ten = pipe(
    BigInt(100),
    clamp (ordBigInt) (-10, 10)
) // return 10n
```

### Suggested Solution

In `Ord.ts`:

```typescript
function looseEqual<A>(a: A, b: A): boolean {
    return a == b
}

export const ordBigInt: Ord<BigInt> = {
    equals: looseEqual,
    compare
}
```

### Who does this impact? Who is this for?

This is for fp-ts users who deal with the `bigint` type and any orderings

### Describe alternatives you've considered

``` typescript
const ordBigInt = contramap ((a: bigint) => a.toString()) // incorrect
```

``` typescript
const ordBigInt = contramap ((a: bigint) => +a.toString()) // overflow risk
```

### Your environment

| Software   | Version(s) |
| ---------- | ---------- |
| fp-ts      |   2.8.5         |
| TypeScript |   3.9.7      |

### PR DESCRIPTION
**Pull request by EricCrosson**
Closes #1371

### CODE REVIEW

**Comment by EricCrosson** [comment]
Open to feedback :+1:

**Comment by pashutk** [comment]
Hi!

Do you have a reason to use loose equality instead of `strictEqual`?
Seems like `===` works well for bigint equality check.

Screenshot 2021-01-15 at 19 35 32

**Comment by EricCrosson** [comment]
I did witness that too in a test after I put up this PR, but came to this implementation after reading the documentation on [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#comparisons). This doc seems to be a superset of the information in the [tc39 bigint proposal](https://github.com/tc39/proposal-bigint#comparisons).

Since `bigint`s are wrapped in objects, this loose equality makes sense from a theoretical level. I would expect the strict equality to return true only when the left and right operand are the same object in memory. I'm not sure how to explain yours and my local test just yet

**Comment by pashutk** [comment]
@EricCrosson 
> I would like some confidence about why the strict equality should be a correct alternative before diverging from the specification.

Maybe this can help: [3.2.6 Strict Equality Comparison](https://tc39.es/proposal-bigint/#sec-strict-equality-comparison)

**Comment by pashutk** [comment]
Also, looks like the link you mentioned as a bigint proposal telling us about comparison between bigint and number, not between two instances of bigint.

File: test/Eq.ts [hunk 1]
```diff
@@ -53,6 +53,12 @@ describe('Eq', () => {
     assert.deepStrictEqual(_.eqDate.equals(new Date(1), new Date(0)), false)
   })
 
+  it('eqDate', () => {
```
Comment (DenisFrezzato) [inline test/Eq.ts:56]:
```suggestion
  it('eqBigInt', () => {
```

**Comment by EricCrosson** [comment]
> Also, looks like the link you mentioned as a bigint proposal telling us about comparison between bigint and number, not between two instances of bigint.

Aha, yes that does clear it up! Thanks for the helpful links @pashutk, I'm much happier to use strict equality

**Comment by gcanti** [comment]
Thanks @EricCrosson @pashutk @DenisFrezzato

**Pull request merged**